### ISSUE TITLE
Slow download of EPG

### ISSUE THREAD

**Issue by Dimension02000** [iptv-org/epg#90]
I am using Kodi 19 along with the add-on iVue for my EPG setup. When I launch iVue for the first time of the day I have it setup to download the EPG information from https://iptv-org.github.io/epg/guides/tvtv.us.guide.xml. For some reason it is taking forever to download the information. I am talking it can easily take 3 minutes to complete the download. 

Is there a way that I can edit down the number of channels that it is downloading so that it does not waste so much time downloading channels that I do not have access to? I tried the other US XML file but it does not have all of the channels I am looking for so I am stuck with the largest one. I have iVue setup to ignore the channels that I do not have access to which took me like an hour to complete.

**Comment by freearhey** [comment]
Please try again using this link this time: `https://iptv-org.github.io/epg/guides/us/tvtv.us.epg.xml`

If it still takes too long to load, try this guide: `https://iptv-org.github.io/epg/guides/us/tvguide.com.epg.xml`. It should load a lot faster, but it only has 88 channels so far.

**Comment by Dimension02000** [comment]
Thanks for getting back to me on this issue but I actually went a different way. Instead of using the Kodi guide to turn on and off the channels I have access to I discovered IPTVBoss. This allowed me to import my m3u file and then I just pointed it to EPG page I wanted. Created a scheduled task and now it is all automated and only downloads what I what. Takes about 10 seconds to get all the data at startup.
